packages/docs/Examples/sale.md

```diff
@@ -5,120 +5,245 @@ icon: flame
 
 # Sales
 
-In the case of a swap, both smart objects have to be known before the swap transaction can be built. In this section we show how a single smart object can be offered for sale at a specified price.
+!!!
+These examples use several advanced features (sighash types, mocking, and controlling the order of inputs and outputs, ordinal safe programming) that are not sufficiently documented elsewhere. If you would like to use these features we suggest to ask about their safe use [here](https://t.me/thebitcoincomputer).
+!!!
 
-## Sale
+In this section we explain how a smart object can be sold at a predetermined price without requiring trust between seller and buyer. Unlike in a swap where both objects are known in advance, the output that the buyer will provide later is unknown at the time of initiating the swap transaction. To address this problem, the Bitcoin Computer offers a feature called 'mocking,' which enables the Seller to create a mock-up of the payment object that the buyer will create later.
 
-We first explain how to execute the sale of a smart object. The smart contract is quite simple but the swap does not preserving ordinal ranges of the smart object being sold. It is therefore not safe to use with ordinals. In the next section we explain a slightly more complicated swap that can be used with ordinals. 
+## Smart Object Sale
 
-### Smart Contract
+We first explain how to execute the sale of a smart object. The smart contract is quite simple but does not preserve the ordinal ranges of the smart object being sold. It is therefore not safe to use with ordinals. The [next](#ordinals-sale) section explains a slightly more complicated swap that can be used with ordinals.
 
-```ts
-class Sale extends Contract {
-  static exec(a: NFT, b: NFT) {
-    const [ownerA] = a._owners
-    const [ownerB] = b._owners
-    a.transfer(ownerB)
-    b.transfer(ownerA)
-    return [b, a]
-  }
-}
-```
+The idea is to build a "crossover" transaction with two inputs and two outputs. The first input of the transaction spends the NFT `n`, and the second input will spend the payment `p`. The first output of the transaction will be `p`, and the second output will be `n` after the swap.
+
+Seller signs the first input and output with the sighash type `SIGHASH_SINGLE | SIGHASH_ANYONECANPAY` so that any other user can modify the second input and output or add arbitrary inputs and outputs as longs as the input-output pair signed by Seller have the same index. As the input-output pair is signed, Seller is guaranteed that if a transaction containing the pair is included in the blockchain then Seller will get paid. 
```

Comment:
Here is the first reference to sighash type, we can link to the docs here.
`
[sighash type](https://developer.bitcoin.org/devguide/transactions.html#signature-hash-types)
`